File: app/src/main/java/com/example/reminders20/RemindersApplication.java
package com.example.reminders20;

import android.app.Application;

import com.example.reminders20.di.DaggerRemindersComponent;
import com.example.reminders20.di.RemindersComponent;

public class RemindersApplication extends Application {

    private static RemindersComponent rootComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        rootComponent = DaggerRemindersComponent.builder().application(this).build();
    }

    public static RemindersComponent getRootComponent() {
        return rootComponent;
    }
}

File: app/src/main/java/com/example/reminders20/RemindersApplication.kt
package com.example.reminders20

import android.app.Application
import androidx.room.Room
import com.example.reminders20.db.ReminderDatabase
import com.example.reminders20.viewModels.ListViewModel
import com.example.reminders20.viewModels.NewReminderViewModel
import org.koin.android.ext.koin.androidContext
import org.koin.android.ext.koin.androidLogger
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.core.context.startKoin
import org.koin.dsl.module

class RemindersApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        val daoModule = module {
            single {
                Room.databaseBuilder(get(), ReminderDatabase::class.java, "reminder_db")
                    .build()
                    .reminderDao()
            }
            single {  }
        }
        val viewModelModule = module {
            viewModel { ListViewModel(get()) }
            viewModel { NewReminderViewModel(get()) }
        }
        startKoin {
            androidLogger()
            androidContext(this@RemindersApplication)
            modules(daoModule, viewModelModule)
        }
    }
}